### ISSUE TITLE
Add Ability to trust individual sever side SSL certs/CA certs

### ISSUE THREAD

**Issue by SCP-682** [hexchat/hexchat#261]
As it stands, hexchat is not currently able to connect to an irc network that does not have a server CA cert in the openssl keyring in a trusted manner, IE the ability to uncheck the "accept invalid cert" box.

I am proposing adding some kind of ability to configure trusting either a CA root key on a network wide basis (works for networks like freenode or OFTC that use a single CA to sign all the keys on all their servers), or trust an individual server's keys in the case of networks like efnet, where every server has its own self signed key.

Id also suggest ignoring the fact a certificate is expired, as many IRC servers use them, and the fact they are expired is not a real measure of security at all, simply verify the keys.

My suggestions are:

either when you first connect to an irc server, it pops up a dialouge asking you if you want to trust the server's key (openssh style), then if it changes in the future, it gives you a warning

or

Have a mechanism for hexchat to read a CA key or self signed server key from a file upon connect, and verify that.

**Comment by kekboy** [comment]
Guys please merge the sslalert branch, it became tested already and it's not a big part of the code to maintain in case there is an issue.
The community really needs this feature.

**Comment by vlanmx** [comment]
Agreed, we need this feature asap and I also can confirm the code in the `sslalert` branch is quite stable.

**Comment by alxpettit** [comment]
It's pretty frustrating that we have gone 6 years without this basic feature being implemented on a high-profile IRC client.

**Comment by tmladek** [comment]
Still a problem...
